### ISSUE TITLE
AUPR value is too low

### ISSUE THREAD

**Issue by NightCrus1ng** [YihePang/DisoFLAG#2]
Hello author, first of all, I would like to express my sincere thanks to you for providing me with your experimental results, as they have been a great impetus to my research on this topic. I have basically reproduced your paper, but when comparing it with Table1 in the article, I found that the AUPR gap is very large, while other metrics such as AUC, Fmax, MCC, and BACC are close to each other. This is a very tricky problem, I found that the dataset categories are extremely unbalanced on the DP93 test set, and the AUPR value is still very low after using a series of methods such as weighting and sampling, and even when using your trained model_730.pkl weights file for the evaluation of the test, the item is still not improved, so I would like to ask if you have encountered this problem at that time or how do you solve the problem of data imbalance. I looking forward to your reply.

![Image](https://github.com/user-attachments/assets/dac743dd-254d-42f8-b3b3-6597d5bdd95e)

![Image](https://github.com/user-attachments/assets/176c2a97-d542-404e-acdd-b4e00d3d0e16)

**Comment by YihePang** [comment]
I strongly recommend that you check the evaluation manner of the AUPR.
If you can confirm that there are no error, then it is highly likely that the problem is due to overfitting. 
During the training process, the optimization was performed using the average loss of disordered and six disordered functions. Here is the  definition of the loss function used in model training:
one_loss_0 = criterion(one_IDR_logits, one_IDP_target)
one_loss_1 = criterion(one_PB_logits, one_PB_target)
one_loss_2 = criterion(one_DB_logits, one_DB_target)
one_loss_3 = criterion(one_RB_logits, one_RB_target)
one_loss_4 = criterion(one_IB_logits, one_IB_target)
one_loss_5 = criterion(one_LB_logits, one_LB_target)
one_loss_6 = criterion(one_Link_logits, one_Link_target)
one_loss = (one_loss_0 + one_loss_1 + one_loss_2 + one_loss_3 + one_loss_4 + one_loss_5 + one_loss_6) / 7

**Comment by NightCrus1ng** [comment]
Thank you for your reply. First of all, to rule out the cause of overfitting, the results of the weights you gave are shown in the figure below. If there may be a certain degree of overfitting during the training process, this should be reflected in the gap between all test set indicators and a certain ideal state, rather than just selectively affecting AUPR and BACC. Secondly, I think there should be no problem with the AUPR evaluation method. The code is as follows. Experiments have found that threshold selection is the most likely cause of BACC differences. My script uses the threshold of optimized Fmax to calculate BACC, but the most puzzling thing is AUPR. This result is really unreasonable.
# evaluate_table1.py (在 evaluate_model_performance 函数内部)

        # ... (省略了获取 y_true 和 y_prob 的代码) ...

        # 检查是否有足够的有效数据和类别
        if len(y_true) > 0 and len(y_prob) == len(y_true) and len(np.unique(y_true)) >= 2:
            # 检查预测值中是否有无效值 (NaN 或 Inf)
            if np.any(np.isnan(y_prob)) or np.any(np.isinf(y_prob)):
                 print(f"警告：功能 '{func_name}' 的预测概率包含 NaN 或 Inf，指标无法计算。")
            else:
                try:
                    auc = roc_auc_score(y_true, y_prob) # 计算 AUC
                except ValueError as e: print(f"计算 '{func_name}' AUC 时出错: {e}")

![Image](https://github.com/user-attachments/assets/034d35bd-cf8e-4f19-b8d3-137369dab217)

                # --- 计算 AUPR 的代码段在这里 ---
                try:
                    # 使用 sklearn.metrics.average_precision_score 计算 AUPR
                    aupr = average_precision_score(y_true, y_prob)
                except ValueError as e:
                    # 如果 y_true 只有单一类别，此函数可能报错
                    print(f"计算 '{func_name}' AUPR 时出错: {e}")
                    aupr = np.nan # 出错时设为 NaN
                # --- AUPR 计算结束 ---

                # 计算 Fmax 及其对应阈值
                fmax, threshold_fmax = calculate_fmax(y_true, y_prob)
                # 使用 Fmax 阈值计算 MCC 和 BACC
                mcc, bacc = calculate_mcc_bacc_at_threshold(y_true, y_prob, threshold_fmax)
        # ... (后续代码，处理数据不足或标签单一的情况) ...

        # 存储结果
        results[func_name] = {'AUC': auc, 'AUPR': aupr, 'Fmax': fmax, 'MCC': mcc, 'BACC': bacc}

    return results

**Comment by YihePang** [comment]
Thank you very much for your reminder. I have identified some issues that occurred during the editing and publication process of the article. The AUPR column in Table 1 should not be appeared like that. 

I have created a new “issue” for this purpose to formally correct Table 1. Please move to this section for further details: https://github.com/YihePang/DisoFLAG/issues/3

**Comment by NightCrus1ng** [comment]
OK, I understand. Thank you.
